core/src/main/java/org/jdbi/v3/core/mapper/EnumMapperFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper;
+
+import java.lang.reflect.Type;
+import java.util.Optional;
+import org.jdbi.v3.core.config.ConfigRegistry;
+
+import static org.jdbi.v3.core.generic.GenericTypes.getErasedType;
+
+/**
+ * Column mapper factory which knows how to map Enums.
+ */
+public class EnumMapperFactory implements ColumnMapperFactory {
+    // TODO relation to EnumByNameMapperFactory?
```

Comment:
I would just use `EnumByNameMapperFactory` rather than introduce a new type--maybe amend that class to use `rawType.asSubclass()` as you've done here

Reply:
I'm not savvy on the logic involved in this function, like the asSubclass() call. Do I replace the logic in EnumMapperByNameFactory with the logic in this class, and then delegate to it? A future improvement would be to make this class configurable to use either ByName or ByOrdinal. I like how this division of responsibilities sounds.